content/docs/user-guide/dvc-files.md

```diff
@@ -129,7 +129,14 @@ can be written manually or generated by user code.
 Here's a comprehensive `dvc.yaml` example:
 
 ```yaml
-stages:
+vars: # Optional section
+  - train:
+      script: train.py
```

Comment:
- [ ] This intro example to dvc.yaml is probably getting too complicated...

Reply:
Should we work on a dvc.yaml writing guide along with or in parallel to this @shcheklein ? That way I can leave only the basic essentials here for now and leave all the possibilities for that separate guide or how-to.

Reply:
@jorgeorpinel, I think it'd be better to not show parametrization right away, and put the note at the end. 

> Should we work on a dvc.yaml writing guide along with?

That'd be great, but even if not, maybe we could separate the dvc.yaml format explanation?
Like `dvc.yaml` short reference, then a section about "stage definition" explaining what key/value it takes, then "variables/parametriation" and then introduce loops. This way, concepts will get built as we go.

Reply:
it's already part of the user guide, right? may be just split it into subsections - one about regular simple pointers .dvc, second about dvc.yaml and start writing a proper guide in it, start building proper structure for it. We can take a look at similar docs (spacy, ansible, etc, @skshetry have you seen some examples that you like?)

Reply:
It's already in the user guide indeed. I meant a how-to, I suppose.

Reply:
OK so on this: I've removed the `vars` section in the first comprehensive example and only left a simple example of `${}` in the 2nd stage (and a note under the code block about that, with a link to the appropriate section).

> dvc.yaml short reference, then a section about "stage definition"...

I like this idea as a first step towards a proper "guide" but I'm not sure it's within this PR's scope.

> just split it into subsections - one about regular simple pointers .dvc, second about dvc.yaml and start writing a proper guide in it

I think for a proper guide we need to write it from scratch because the current doc from the beginning is much more of a reference (without a story). In any case I've extracted this into a separate issue (which is not 2.0 specific but we can still prioritize) to not complicate this PR too much: #2059